app/src/main/kotlin/com/intive/picover/auth/repository/AuthRepository.kt

```diff
@@ -30,10 +34,23 @@ class AuthRepository @Inject constructor(
 		firebaseAuth.signOut()
 	}
 
+	suspend fun userProfile() = withContext(ioDispatcher) {
```

Comment:
Not blocking comment :D: Is the IO dispatcher really needed here? I'm just wondering if there was some reason why you added it?

Reply:
Back to the comment https://github.com/intive/Picover/pull/139#discussion_r1251129071and considering the current, I think at this moment I didn't need coroutines here and use `viewModelScope.launch` in method `fetchProfile()` in `ProfileViewModel` probably I will add it or change something when I will implement update avatar photo and other profile data. 

So it will look like this: 

`AuthRepository`
```
	fun userProfile() =
		requireUser().let {
			Profile(
				photo = it.photoUrl,
				name = it.displayName!!,
				email = it.email!!,
			)
		}
```
`ProfileViewModel`
```
	private fun fetchProfile() {
		_profile.value = Loaded(authRepository.userProfile())
	}
```
What do you think of it @BorysChajdas, @marcinWisnia

Reply:
Now I changed it